CSCfi/swift-browser-ui
# Feature/redirect existing sessions

### Description

Valid logged-in sessions were previously non-persistent. I.e. people with open sessions were forced to log in again instead of being redirected to the UI. This behavior has now been fixed, and established sessions are persistent with a inaction period of 1h and a total session lifetime of 8h, matching that of Openstack.

### Related issues

* Implements / fixes issue #111 

### Type of change




- [ ] This change requires a documentation update (this needs a follow up PR)

### Changes Made


* Make sessions persistent during lifetime
* Refactor session invalidation into a function
* Check for session lifetime instead of trusting Openstack to invalidate in time
* Remove separate set for session ids as unnecessary
* Refactor tests to account for session redirection
* Reduce token length from 512 bits to 256 as previous was unnecessarily long and the shorter one is secure enough
* Change tox *flake8* rules in preparation for moving to *black* coding style

### Testing





- [x] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### Mentions
